apps/src/sites/studio/pages/jotformLoader.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file Entry point for a bundle to embed on pages where we are also letting JotForm embed
+ * an inline form using an iframe.  This bundle monitors the JotForm script and attempts
+ * to measure reachability of the JotForm domain, load success, load time, and reports metrics
+ * back to our system so we have a sense of how often this fails.
+ * We also attempt to improve the user experience when the form is unavailable.
+ */
+import logToCloud from '../../../logToCloud';
+
+function main(context) {
+  Promise.all([
+    checkJotFormFrameLoaded(context),
+    // Domains used by Jotform:
+    // cdn.jotfor.ms       https://cdn.jotfor.ms/images/calendar.png  817B
+    checkReachability('https://cdn.jotfor.ms/images/calendar.png'),
+    // www.jotform.com     https://www.jotform.com/favicon.ico        887B
+    checkReachability('https://www.jotform.com/favicon.ico')
+    // Not using these yet, they're likely to fall under the same policy as www.jotform.com
+    // events.jotform.com
+    // files.jotform.com
+    // form.jotform.com
+  ]).then(([jotFormFrameLoadedMs, cdnjotformsMs, wwwjotformcomMs]) => {
+    if (jotFormFrameLoadedMs === false) {
+      // Load failed if we specifcially got 'false'
+      logToCloud.addPageAction(logToCloud.PageAction.JotFormLoadFailed, {
+        route: `GET ${context.location.pathname}`,
+        reachedCdnjotforms: false !== cdnjotformsMs,
+        reachedWwwjotformcom: false !== wwwjotformcomMs,
+        cdnjotformsMs,
+        wwwjotformcomMs
+      });
+    } else {
+      logToCloud.addPageAction(logToCloud.PageAction.JotFormFrameLoaded, {
+        route: `GET ${context.location.pathname}`,
+        jotFormFrameLoadedMs,
+        cdnjotformsMs,
+        wwwjotformcomMs
+      });
+    }
+  });
+}
+
+/**
+ * Report page time when JotForm's embedded form successfully loads, or resolve(false) if it
+ * fails to load within five seconds.
+ * @param {Window} context
+ * @returns {Promise<number|false>} Always resolves, page time (ms) if load succeeded, `false` if not.
+ */
+function checkJotFormFrameLoaded(context) {
+  return new Promise(resolve => {
+    context.JotFormFrameLoaded = function() {
+      const time = performance.now();
+      if (timeoutKey) {
+        clearTimeout(timeoutKey);
+      }
+      console.log(`JotFormFrameLoaded fired at ${time}ms`);
```

Comment:
Intentionally leaving `console.log` statements in, since we may be able to ask teachers to pop open their dev console in the future and this information might be helpful.